src/main/java/com/esri/samples/na/routing_around_barriers/RoutingAroundBarriersController.java

```diff
@@ -0,0 +1,335 @@
+  @FXML private TitledPane routeInformationTitledPane;
+
+  private GraphicsOverlay routeGraphicsOverlay;
+  private SimpleLineSymbol routeLineSymbol;
+  private RouteTask routeTask;
+  private RouteParameters routeParameters;
+  private ArrayList<Stop> stopsList;
+  private ArrayList<PolygonBarrier> barriersList;
+
+  @FXML
+  public void initialize() {
+    // create an ArcGISMap with a streets basemap
+    ArcGISMap map = new ArcGISMap(Basemap.createStreets());
+    // set the ArcGISMap to be displayed in this view
+    mapView.setMap(map);
+    // zoom to viewpoint
+    mapView.setViewpoint(new Viewpoint(32.727, -117.1750, 40000));
+
+    // create graphics overlays for stops, barriers and route
+    GraphicsOverlay stopsGraphicsOverlay = new GraphicsOverlay();
+    GraphicsOverlay barriersGraphicsOverlay = new GraphicsOverlay();
+    routeGraphicsOverlay = new GraphicsOverlay();
+
+    // add the graphics overlays to the map view
+    mapView.getGraphicsOverlays().addAll(Arrays.asList(stopsGraphicsOverlay, barriersGraphicsOverlay, routeGraphicsOverlay));
+
+    // create a list of stops and a list of barriers
+    stopsList = new ArrayList<>();
+    barriersList = new ArrayList<>();
+
+    // initialize the TitlePane of the Accordion box
+    routeInformationTitledPane.setText("No route to display");
+
+    // create symbols for displaying the barriers and the route line
+    SimpleFillSymbol barrierSymbol = new SimpleFillSymbol(SimpleFillSymbol.Style.DIAGONAL_CROSS, 0xFFFF0000, null);
+    routeLineSymbol = new SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, 0x800000FF, 5.0f);
+
+    // create route task from San Diego service
+    routeTask = new RouteTask("http://sampleserver6.arcgisonline.com/arcgis/rest/services/NetworkAnalysis/SanDiego/NAServer/Route");
+    routeTask.loadAsync();
+
+    // wait for the route task to load
+    routeTask.addDoneLoadingListener(() -> {
+      if (routeTask.getLoadStatus() == LoadStatus.LOADED) {
+        // get default route parameters
+        final ListenableFuture<RouteParameters> routeParametersFuture = routeTask.createDefaultParametersAsync();
+        routeParametersFuture.addDoneListener(() -> {
+          try {
+            routeParameters = routeParametersFuture.get();
+
+            // set flags to return stops and directions
+            routeParameters.setReturnStops(true);
+            routeParameters.setReturnDirections(true);
+
+            // enable the UI
+            btnDetermineRoute.setDisable(false);
+
+          } catch (InterruptedException | ExecutionException e) {
+            new Alert(Alert.AlertType.ERROR, "Cannot create RouteTask parameters " + e.getMessage()).show();
+          }
+        });
+
+      } else {
+        new Alert(Alert.AlertType.ERROR, "Unable to load RouteTask " + routeTask.getLoadStatus().toString()).show();
+      }
+    });
+
+    // listen to mouse clicks to add/remove stops or barriers
+    mapView.setOnMouseClicked(e -> {
```

Comment:
Could separate this out into another method and reference it in the fxml with
```xml
<MapView fx:id="mapView" onMouseClicked="#handleMapViewClicked"/>
```

Reply:
Good call, done!